Assets/Scripts/Player/UnitGridCombat.cs

```diff
@@ -178,6 +182,10 @@ private enum Direction {
 				return;
 			}
 			DetermineAttackAnimation(unitGridCombat);
+			audioSource_.clip = weapon.GetAttackSound();
+			if(audioSource_.clip != null) {

```

Comment:
Just one question, when is the `audioSource_.clip` ever set to `null`, I get the check but just wondering when this case comes up.

Reply:
When we don't set any audio for the weapon at all, so yeah this can be null.